cmd/podman/containers/create.go

```diff
@@ -324,3 +339,20 @@ func createPodIfNecessary(s *specgen.SpecGenerator, netOpts *entities.NetOptions
 	s.Pod = podName
 	return registry.ContainerEngine().PodCreate(context.Background(), createOptions)
 }
+
+// inheritPodUserNS gets the userns option that the pod was created with and sets that
+// for the container being created (if the container is being created inside a pod).
+func inheritPodUserNS(s *specgen.SpecGenerator) error {
+	if s.Pod == "" || strings.HasPrefix(s.Pod, "new:") {
+		return nil
+	}
+	inspectOptions := entities.PodInspectOptions{
+		NameOrID: s.Pod,
+	}
+	data, err := registry.ContainerEngine().PodInspect(context.Background(), inspectOptions)
+	if err != nil {
+		return errors.Errorf("couldn't inspect pod %q", s.Pod)
+	}
+	s.UserNS, err = specgen.ParseUserNamespace(data.InfraConfig.UserNS)
```

Comment:
should the effect be something like `WithUserNSFrom()` and join the user namespace for the infra container?